tests/Kernel/Containers/iterator_test.cpp

```diff
@@ -68,4 +90,26 @@ TEST_CASE ("iterate on hashset") {
     a << as_string (it->next ());
   }
   CHECK_EQ (a == string ("123"), true);
+}
+
+TEST_CASE ("ranged for on hashset") {
+  SUBCASE ("ordinary") {
+    hashset<int> h;
+    h->insert (1);
+    h->insert (2);
+    h->insert (3);
+    string a;
+    for (auto i : iterate (h)) {
+      a << as_string (i);
+    }
+    string_eq (a, "123");
```

Comment:
Are the elements in our hashmap arranged in insertion order?

Reply:
According to prior tests in this file, orders is arranged as you said.